cmd/syncthing/main.go

```diff
@@ -358,15 +360,31 @@ func main() {
 		return
 	}
 
+	checkUpgradeExitOnErr := func() upgrade.Release {
```

Comment:
I think this should be a normal function despite the given rationale. Closures have a significant mental cost, at least for me, which is not necessary in this case

Reply:
Or maybe just inline this and have a function `exitCodeForUpgrade(error) int` that does the error checking and returns the appropriate exit code, as that's the only real logic, and do the exit inline in main